src/main/kotlin/org/rust/cargo/toolchain/Rustfmt.kt

```diff
@@ -26,38 +29,34 @@ class Rustfmt(private val rustfmtExecutable: Path) {
         val file = document.virtualFile ?: return null
         if (file.isNotRustFile || !file.isValid) return null
 
-        val configPath = findConfigPathRecursively(file.parent, stopAt = cargoProject.workingDirectory)
-        return reformatText(cargoProject, document.text, configPath)
-    }
-
-    @Throws(ExecutionException::class)
-    private fun reformatText(
-        cargoProject: CargoProject,
-        text: String,
-        configPath: Path? = null,
-        parentDisposable: Disposable = cargoProject.project,
-        skipChildren: Boolean = checkSupportForSkipChildrenFlag(cargoProject)
-    ): String? {
         val arguments = buildList<String> {
             add("--emit=stdout")
 
-            if (skipChildren) {
+            if (checkSupportForSkipChildrenFlag(cargoProject)) {
                 add("--unstable-features")
                 add("--skip-children")
             }
 
+            val configPath = findConfigPathRecursively(file.parent, stopAt = cargoProject.workingDirectory)
             if (configPath != null) {
                 add("--config-path")
                 add(configPath.toString())
             }
+
+            val currentRustcVersion = cargoProject.rustcInfo?.version?.semver
+            if (currentRustcVersion != null && currentRustcVersion >= RUST_1_31) {
+                val psiFile = runReadAction { file.toPsiFile(cargoProject.project) }
+                val isEdition2018 = psiFile?.isEdition2018 ?: true
+                add("--edition=${if (isEdition2018) "2018" else "2015"}")
```

Comment:
I think it's worth to add some method for `Edition` to convert it to a year-like string.
After that, this code will work with arbitrary future edition